Src/VimMac/AlternateKeyUtil.cs

```diff
@@ -0,0 +1,256 @@
+                [VimKey.F5] = NSKey.F5,
+                [VimKey.F6] = NSKey.F6,
+                [VimKey.F7] = NSKey.F7,
+                [VimKey.F8] = NSKey.F8,
+                [VimKey.F9] = NSKey.F9,
+                [VimKey.F10] = NSKey.F10,
+                [VimKey.F11] = NSKey.F11,
+                [VimKey.F12] = NSKey.F12,
+                [VimKey.KeypadMultiply] = NSKey.KeypadMultiply,
+                [VimKey.KeypadPlus] = NSKey.KeypadPlus,
+                [VimKey.KeypadMinus] = NSKey.KeypadMinus,
+                [VimKey.KeypadDecimal] = NSKey.KeypadDecimal,
+                [VimKey.KeypadDivide] = NSKey.KeypadDivide,
+                [VimKey.KeypadEnter] = NSKey.KeypadEnter,
+                [VimKey.Keypad0] = NSKey.Keypad0,
+                [VimKey.Keypad1] = NSKey.Keypad1,
+                [VimKey.Keypad2] = NSKey.Keypad2,
+                [VimKey.Keypad3] = NSKey.Keypad3,
+                [VimKey.Keypad4] = NSKey.Keypad4,
+                [VimKey.Keypad5] = NSKey.Keypad5,
+                [VimKey.Keypad6] = NSKey.Keypad6,
+                [VimKey.Keypad7] = NSKey.Keypad7,
+                [VimKey.Keypad8] = NSKey.Keypad8,
+                [VimKey.Keypad9] = NSKey.Keypad9
+            };
+
+            return map;
+        }
+
+        internal static Dictionary<KeyInput, NSKey> BuildKeyInputToCocoaKeyMap(Dictionary<VimKey, NSKey> vimKeyToCocoaKeyMap)
+        {
+            var map = new Dictionary<KeyInput, NSKey>();
+            foreach (var pair in vimKeyToCocoaKeyMap)
+            {
+                var keyInput = KeyInputUtil.VimKeyToKeyInput(pair.Key);
+                map[keyInput] = pair.Value;
+            }
+
+            map[KeyInputUtil.CharToKeyInput(' ')] = NSKey.Space;
+            map[KeyInputUtil.CharToKeyInput('\t')] = NSKey.Tab;
+
+            return map;
+        }
+
+        internal static bool TrySpecialVimKeyToKey(VimKey vimKey, out NSKey key)
+        {
+            return s_vimKeyToCocoaKeyMap.TryGetValue(vimKey, out key);
+        }
+
+        internal static VimKeyModifiers ConvertToKeyModifiers(NSEventModifierMask keys)
+        {
+            var res = VimKeyModifiers.None;
+            if (keys.HasFlag(NSEventModifierMask.ShiftKeyMask))
+            {
+                res |= VimKeyModifiers.Shift;
+            }
+            if (keys.HasFlag(NSEventModifierMask.AlternateKeyMask))
+            {
+                res |= VimKeyModifiers.Alt;
+            }
+            if (keys.HasFlag(NSEventModifierMask.ControlKeyMask))
+            {
+                res |= VimKeyModifiers.Control;
+            }
+            if (keys.HasFlag(NSEventModifierMask.CommandKeyMask))
+            {
+                res |= VimKeyModifiers.Command;
+            }
+            return res;
+        }
+
+        internal static NSEventModifierMask ConvertToModifierKeys(VimKeyModifiers keys)
+        {
+            NSEventModifierMask res = 0;
+            if (0 != (keys & VimKeyModifiers.Shift))

```

Comment:
Consistency: Use HasFlags as above. Mono/.netcore have runtime instrincs for them, so both versions will generate the same code